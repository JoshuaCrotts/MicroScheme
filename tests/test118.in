;
; LCM calculation using GCD.
;

(define abs (lambda (x) (if (<= x 0) (* x -1) x)))
(define zero? (lambda (x) (= x 0)))

(define gcd (lambda (a b)
    (if (zero? b)
        a
        (gcd b (modulo a b)))))
(define lcm (lambda (a b)
    (if (or (zero? a) (zero? b))
        0
        (abs (* b (truncate (/ a (gcd a b))))))))

(lcm 12 18)
(lcm 330 75)
(lcm 16850 221625)
(lcm 24 41525)
