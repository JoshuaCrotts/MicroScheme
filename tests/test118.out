36
1650
149375250
996600
